### ISSUE TITLE
[css-color-4] Clarify interactions with host syntax

### ISSUE THREAD

**Issue by sesse** [w3c/csswg-drafts#7567]
The spec goes to some length about what host syntax is, and specifies: “These features are collectively termed the host syntax. The host syntax should define what the default interpolation space should be for each case, and optionally provide syntax for authors to override this default. If such syntax is part of a property value, it should use a <color-interpolation-method> token, defined below for easy reference from other specifications. ”

However, it's not clear at all what “the host syntax should define” means. Does this mean that the spec expects e.g. that the CSS animations spec be changed to have new CSS properties for interpolation color space? (And similar for gradients and filters.) There is new syntax in §12.1 (“Color Space for Interpolation”), but it doesn't seem to be connected to anything in the larger scope of CSS. Is something missing from the color spec, or will other specs fill this in at some future date?

Concretely, if I as an author would want to make a gradient that's interpolating using (say) HSL along the longer axis, how would I express that? Can I do that at all currently?

**Comment by lilles** [comment]
@svgeesus @tabatkins @LeaVerou

**Comment by faceless2** [comment]
css-images-4 defines a syntax for gradient interpolation in a particular space - see eg https://drafts.csswg.org/css-images-4/#linear-gradients

**Comment by svgeesus** [comment]
> However, it's not clear at all what “the host syntax should define” means. Does this mean that the spec expects e.g. that the CSS animations spec be changed to have new CSS properties for interpolation color space?

Yes. "should" define because currently it hard-codes gamma-encoded sRGB and does not allow any other option.

>  (And similar for gradients and filters.) 

Yes. Gradients already does, as @faceless2 pointed out.

**Comment by svgeesus** [comment]
> Concretely, if I as an author would want to make a gradient that's interpolating using (say) HSL along the longer axis, how would I express that? Can I do that at all currently?

```css
linear-gradient(in hsl longer to right, #F01, #081);
```

**Comment by svgeesus** [comment]
Hi @sesse did these replies answer your question?

**Comment by sesse** [comment]
It did, but it would be nice to have explicit references in the text, perhaps? Something like “these declarations are not used by this spec itself, only exposed so that other specifications can use them; see e.g. use in [CSS-IMAGES-4]#reference”?

**Comment by svgeesus** [comment]
Ah, fair enough. CSS being a modular specification, it is common for one module to expose things so that other modules can do the same thing consistently, but  will add wording to that effect here.
